### ISSUE TITLE
ArrayIndexOutOfBounds in SdlRouterService when connecting to HU.

### ISSUE THREAD

**Issue by jthrun** [smartdevicelink/sdl_android#350]
This happens only some of the time, seems like after HS/phone restarted then first connect to HU.

https://github.com/smartdevicelink/sdl_android/blob/master/sdl_android_lib/src/com/smartdevicelink/transport/SdlRouterService.java#L1800

I'm getting this stack trace:

```
10-25 19:04:14.393 8674-8674/com.smartdevicelink.router E/AndroidRuntime: FATAL EXCEPTION: main
        Process: com.smartdevicelink.router, PID: 8674
        java.lang.ArrayIndexOutOfBoundsException: src.length=83 srcPos=0 dst.length=83 dstPos=12 length=83
        at java.lang.System.arraycopy(Native Method)
        at com.smartdevicelink.transport.SdlRouterService.createForceUnregisterApp(SdlRouterService.java:1800)
        at com.smartdevicelink.transport.SdlRouterService.attemptToCleanUpModule(SdlRouterService.java:1398)
        at com.smartdevicelink.transport.SdlRouterService.removeAllSessionsForApp(SdlRouterService.java:1674)
        at com.smartdevicelink.transport.SdlRouterService.access$900(SdlRouterService.java:77)
        at com.smartdevicelink.transport.SdlRouterService$RouterHandler.handleMessage(SdlRouterService.java:364)
        at android.os.Handler.dispatchMessage(Handler.java:102)
        at android.os.Looper.loop(Looper.java:135)
        at android.app.ActivityThread.main(ActivityThread.java:5221)
        at java.lang.reflect.Method.invoke(Native Method)
        at java.lang.reflect.Method.invoke(Method.java:372)
        at com.android.internal.os.ZygoteInit$MethodAndArgsCaller.run(ZygoteInit.java:899)
        at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:694)
```

I added a log message to see what the version parameter was:

`10-26 10:33:11.004  6098  6098 E JT_DBG_RTR: Going to crash with ArrayIndexOutOfBounds on messsage of version -1`

**Comment by joeygrover** [comment]
Looks like it includes the binary frame header even for v1 packets which do not have them. Which would then cause the overrun.

**Comment by jthrun** [comment]
I see the 4.2.0 tag, do you know when this will merge to master?  If it will be a while can we request this be hotfix to master 4.1.x?

**Comment by joeygrover** [comment]
Fixed with #351. 4.2.0 is scheduled for 11/11
